src/PhpParser/AstResolver.php

```diff
@@ -340,15 +331,34 @@ public function resolvePropertyFromPropertyReflection(
         return $this->findPromotedPropertyByName($nodes, $desiredPropertyName);
     }
 
+    private function locateClassMethodInTrait(string $className, string $methodName): ?Node
+    {
+        if (! $this->reflectionProvider->hasClass($className)) {
+            return null;
+        }
+
+        $classReflection = $this->reflectionProvider->getClass($className);
+        if ($classReflection->isBuiltIn()) {
+            return null;
+        }
+
+        $traits = $this->parseClassReflectionTraits($classReflection);
+
+        return $this->betterNodeFinder->findFirst(
+            $traits,
+            fn (Node $node): bool => $node instanceof ClassMethod && $this->nodeNameResolver->isName($node, $methodName)
+        );
+    }
+
     /**
-     * @return Stmt[]|null
+     * @return Stmt[]
      */
-    private function parseFileNameToDecoratedNodes(string $fileName): ?array
+    private function parseFileNameToDecoratedNodes(string $fileName): array
     {
         $fileContent = $this->smartFileSystem->readFile($fileName);
         $nodes = $this->parser->parse($fileContent);
         if ($nodes === null) {
-            return null;
+            return [];
         }
 
```

Comment:
Why this change?

I would prefer to keep it, as we had lot of bugs with false positives like these, when we parsed a broken file. Instead of an error we returned nodes with assumption they should be iterated and file is editable.

Returning `null` kills this possibility right in the start.

Reply:
That was to remove unneeded check when call it, used in `findFirst()` and `findInstanceOf()` which pass an array, pass empty array will make return null, which should be ok. 

Ok, I will reverted it https://github.com/rectorphp/rector-src/pull/558/commits/d9311e71371b2809c8a756ebd1667a7a0866b76a